web-client/src/ducks/offers/actions.ts

```diff
@@ -56,10 +58,14 @@ export const setOffer = (payload: Offer | IOffer, offerId?: string) => (
       firebase: offerId ? setUserOffer : createUserOffer,
     });
   } else {
+    const offerPayload = {
```

Comment:
the else condition here means that payload is already an instance of Offer.
So, you must set the value of `seenAt` as `payload.seenAt = firestore.Timestamp.now()` and shouldn't be creating a new object with it.
even if you are creating a new object (which you shouldn't), you should be creating it as: 
`Offer.factory({ ...payload.toObject(), seenAt: firestore.Timestamp.now() })`

Reply:
+1

Reply:
Got it, got this fixed now